src/OrbitGl/Track.h

```diff
@@ -99,7 +100,7 @@ class Track : public orbit_gl::CaptureViewElement, public std::enable_shared_fro
 
   [[nodiscard]] virtual bool IsTrackSelected() const { return false; }
 
-  [[nodiscard]] bool IsCollapsed() const { return collapse_toggle_->IsCollapsed(); }
+  [[nodiscard]] virtual bool IsCollapsed() const { return collapse_toggle_->IsCollapsed(); }
```

Comment:
We override this method in `BasicPagefaultTrack` to return `collapse_toggle_->IsCollapsed() || GetParent()->IsCollapsed()`.

This is because we want to show the collapsed version of major pagefault subtrack (i.e., a `BasicPagefaultTrack` with reduced track height, no legends and no annotations) if the `PagefaultTrack` is collapsed. Hence, we want the major pagefault subtrack to be considered as collapsed if its parent track is collapsed.

Reply:
Okay, I see. In the GpuTrack, we flipped the logic, and the parent is responsible to detect that the child or itself is collapsed. I think both is fine. At some point we should decide one one of those. For now, I guess it is fine.

Reply:
I think that I like more the parent to be responsible, children shouldn't care about the state of the parent. Parent should call or not call draw to their children

Reply:
In the current `PagefaultTrack`, we do let the parent (i.e., the `PagefaultTrack`) to determine call or not call the draw method of children. But the major pagefault has both collapsed version and expanded version. And `IsCollapsed` is used to determine which version we would like to draw. If the children does not know the parent's state, then we need to pass a boolean to the children's draw method to indicate which version we would like to draw.

Reply:
Yes, you are right. Thanks for pointing out. Is it difficult to pass that pointer?